src/components/Form/FormDatePicker.react.js

```diff
@@ -0,0 +1,176 @@
+// @flow
+
+import * as React from "react";
+
+import FormSelect from "./FormSelect.react";
+import FormInputGroup from "./FormInputGroup.react";
+
+type Props = {|
+  children?: React.Node,
+  className?: string,
+  defaultDate: Date,
+  minYear: number,
+  maxYear: number,
+  format: string,
+  monthLabels: Array<string>,
+  onChange?: Function,
+|};
+
+type State = {|
+  currentDate: Date,
+|};
+
+type ChangeTypes = "mm" | "yyyy" | "dd";
+
+class FormDatePicker extends React.PureComponent<Props, State> {
+  static defaultProps = {
+    monthLabels: [
+      "January",
+      "February",
+      "March",
+      "April",
+      "May",
+      "June",
+      "July",
+      "August",
+      "September",
+      "October",
+      "November",
+      "December",
+    ],
+    minYear: 1897,
+    maxYear: new Date().getFullYear(),
+    format: "mm/dd/yyyy",
+    defaultDate: new Date(),
+  };
+
+  state = {
+    currentDate: new Date(),
+  };
+
+  // Set the default date from props
+  componentWillMount(): void {
```

Comment:
This seems somewhat repetitive given you set the `currentDate` in `state = {...` above.  I think we should just pick one